behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
+        } else if (searchType === "artists") {
+            navigate("/" + UrlBasePath.ARTISTS + "/?search=" + inputValue);
+        }
+        setFilteredSuggestions([]);
+    };
+
+    const onClickInput = () => {
+        // if it has any input, display suggestions
+        if (inputValue !== "") {
+            // Filter suggestions based on input value
+            const filteredSuggestions = fullSuggestions.filter(suggestion => {
+                // filter suggestions based on main or sub
+                const matchMain = suggestion.main.toLowerCase().includes(inputValue.toLowerCase());
+                const matchSub = suggestion.sub ? suggestion.sub.toLowerCase().includes(inputValue.toLowerCase()) : false;
+                return matchMain || matchSub;
+            });
+
+            setFilteredSuggestions(filteredSuggestions);
+        } else {
+            setFilteredSuggestions([]);
+        }
+    }
+
+    // handler for when user clicks OFF the search bar input (hide the suggestions)
+    const onBurInput = (event) => {
+        // get the element that user has clicked on
+        const clickTarget = event.relatedTarget;
+
+        // check if it is a suggestion (if it is a suggestion do NOT clear the list because then the suggestion
+        // will disappear before click can be registered
+        if (clickTarget && clickTarget.getAttribute("name") === "suggestion-list-item") {
+            return;
+        }
+        // in all other cases, empty/hide suggestions
+        setFilteredSuggestions([]);
+    }
+
+    // this function updates the state when input changes
+    const handleInputChange = (event) => {
+        setInputValue(event.target.value);
+    };
+
+    // update suggestions on input change
+    useEffect(() => {
+        if (inputValue !== '') {
+            // if a suggestion has been selected, clear the list of suggestions and do the search call (reload results)
+            if (selectedSuggestions) {
+                setFilteredSuggestions([]);
+                setSelectedSuggestions(false) // reset state
+                handleButtonClickOrSubmit()
+            } else {
+                // If a suggestion has NOT been selected, filter suggestions based on input value
+                const filteredSuggestions = fullSuggestions.filter(suggestion => {
+                    // filter suggestions based on if it matches either main or sub criteria
+                    const matchMain = suggestion.main.toLowerCase().includes(inputValue.toLowerCase());
+                    const matchSub = suggestion.sub ? suggestion.sub.toLowerCase().includes(inputValue.toLowerCase()) : false;
+                    return matchMain || matchSub;
+                });
+                setFilteredSuggestions(filteredSuggestions);
+            }
+        } else {
+            setFilteredSuggestions([]);
+        }
+    }, [inputValue])
+
+    // on search type change
+    useEffect(() => {
```

Comment:
Same here, I don't think you need a useEffect here, it could just be done in the search type change handler